docs/reading_events.md

```diff
@@ -0,0 +1,151 @@
+
+## Reading from a specific stream
+
+The simplest way to read a stream forwards is to supply a `PgEventstore::Stream` object.
+
+```ruby
+stream = PgEventstore::Stream.new(context: 'MYAwesomeContext', stream_name: 'User', stream_id: '1')
+PgEventstore.client.read(stream)
+# => [#<PgEventstore::Event 0x1>, #<PgEventstore::Event 0x1>, ...]
+```
+
+### max_count
+
+You can provide the `:max_count` option. This option determines how much records to return in a response. Default is `1000` and it can be changed via `:max_count` configuration setting(see [**"Configuration"**](configuration.md) chapter):
+
+```ruby
+stream = PgEventstore::Stream.new(context: 'MYAwesomeContext', stream_name: 'User', stream_id: '1')
+PgEventstore.client.read(stream, options: { max_count: 100 })
+```
+
+### resolve_link_tos
+
+When using projections(Event Sourcing Projections are meant here) to create new events you can set whether the generated events are pointers to existing events. Setting this value to `true` tells `pg_eventstore` to return the original event instead "link" event.
+
+```ruby
+stream = PgEventstore::Stream.new(context: 'MYAwesomeContext', stream_name: 'User', stream_id: '1')
+PgEventstore.client.read(stream, options: { resolve_link_tos: true })
+```
+
+### from_revision
+
+You can define from which revision number you would like to start to read events:
+
+```ruby
+stream = PgEventstore::Stream.new(context: 'MYAwesomeContext', stream_name: 'User', stream_id: '1')
+PgEventstore.client.read(stream, options: { from_revision: 2 })
+```
+
+### direction
+
+As well as being able to read a stream forwards you can also go backwards. This can be achieved by providing the `:direction` option:
+
+```ruby
+stream = PgEventstore::Stream.new(context: 'MYAwesomeContext', stream_name: 'User', stream_id: '1')
+PgEventstore.client.read(stream, options: { direction: 'Backwards' })
+```
+
+## Checking if stream exists
+
+In case a stream with given name does not exist - `PgEventstore::StreamNotFoundError` error will be raised:
+
+```ruby
+begin
+  stream = PgEventstore::Stream.new(context: 'non-existing-context', stream_name: 'User', stream_id: '1')
+  PgEventstore.client.read(stream)
+rescue PgEventstore::StreamNotFoundError => e
+  puts e.message # => Stream #<PgEventstore::Stream:0x01> does not exist.
+  puts e.stream # => #<PgEventstore::Stream:0x01>
+end
+```
+
+## Reading from the "all" stream
+
+"all" stream definition means that you don't scope your events when reading them from the database. To get "all" `PgEventstore::Stream` instance you have to call special method: 
+
+```ruby
+PgEventstore::Stream.all_stream
+```
+
+Now you can use it to read from "all" stream:
+
+```ruby
+PgEventstore.client.read(PgEventstore::Stream.all_stream)
+```
+
+You can read from a specific position of "all" stream. This is very similar to reading from a specific revision of a specific stream, but instead `:from_revision` option you have to provide `:from_position` option:
```

Comment:
```suggestion
You can read from a specific position of the "all" stream. This is very similar to reading from a specific revision of a specific stream, but instead of the `:from_revision` option you have to provide the `:from_position` option:
```